### ISSUE TITLE
Improve watermark accuracy

### ISSUE THREAD

**Issue by fpgaminer** [fpgaminer/joycaption#19]
JoyCaption should have better watermark detection accuracy, and always mention watermarks if present (unless instructed not to).

TODO: Train a secondary watermark detection model.

TODO: Update JoyCaption's dataset using secondary model and add some watermark focused VQA data.

**Comment by fpgaminer** [comment]
Watermark detection model is in-progress, currently at 93% accuracy.

**Comment by mashdragon** [comment]
How do you usually train a secondary model? Would it simply be a binary image classification model for this example?

**Comment by fpgaminer** [comment]
> How do you usually train a secondary model? Would it simply be a binary image classification model for this example?

Yeah, I'm manually tagging images and then training a classifier head on top of OWLv2's vision model (since it has a 1024x1024 resolution).  I experimented a bit with doing bounding boxes but there's no existing training code for OWL and YOLO is too small of a model to learn quickly off the number of examples I have.  So far I've only been able to get YOLO up to 90% accurate.

**Comment by fpgaminer** [comment]
Complete. 95% watermark detection accuracy model trained. JoyCaption's dataset updated.

(Notes on watermark detection mode: https://civitai.com/articles/10464/wacky-misadventures-in-detecting-watermarks)
